app/src/main/java/com/babylon/wallet/android/data/repository/cache/database/StateDao.kt

```diff
@@ -74,6 +72,8 @@ interface StateDao {
         }.flatten()
         insertOrIgnoreResources(resourcesInvolvedInPools)
 
+        insertPoolDetails(pools.map { it.pool })
```

Comment:
This fixes the foreign key constraint violation exception. In short, the resources involved in the pool we were trying to store didn't exist in the db, so we have first to insert the resources, then the pools. I think we didn't have this kind of situation up until the hidden resources feature, that's why it wasn't an issue before.